package treeni.test;
// Generated by ComTest BEGIN
import treeni.*;
import java.util.*;
import static org.junit.Assert.*;
import org.junit.*;
// Generated by ComTest END

/**
 * Test class made by ComTest
 * @version 2023.01.07 18:56:23 // Generated by ComTest
 *
 */
@SuppressWarnings({ "all" })
public class SuorituksetTest {



  // Generated by ComTest BEGIN
  /** 
   * testLisaa46 
   * @throws SailoException when error
   */
  @Test
  public void testLisaa46() throws SailoException {    // Suoritukset: 46
    Suoritukset suoritukset = new Suoritukset(); 
    Suoritus treeni1 = new Suoritus(), treeni2 = new Suoritus(); 
    assertEquals("From: Suoritukset line: 50", 0, suoritukset.getLkm()); 
    suoritukset.lisaa(treeni1); assertEquals("From: Suoritukset line: 51", 1, suoritukset.getLkm()); 
    suoritukset.lisaa(treeni2); assertEquals("From: Suoritukset line: 52", 2, suoritukset.getLkm()); 
    suoritukset.lisaa(treeni1); assertEquals("From: Suoritukset line: 53", 3, suoritukset.getLkm()); 
    assertEquals("From: Suoritukset line: 54", treeni1, suoritukset.anna(0)); 
    assertEquals("From: Suoritukset line: 55", treeni2, suoritukset.anna(1)); 
    assertEquals("From: Suoritukset line: 56", treeni1, suoritukset.anna(2)); 
    assertEquals("From: Suoritukset line: 57", false, suoritukset.anna(1) == treeni1); 
    assertEquals("From: Suoritukset line: 58", true, suoritukset.anna(1) == treeni2); 
    try {
    assertEquals("From: Suoritukset line: 59", treeni1, suoritukset.anna(3)); 
    fail("Suoritukset: 59 Did not throw IndexOutOfBoundsException");
    } catch(IndexOutOfBoundsException _e_){ _e_.getMessage(); }
    suoritukset.lisaa(treeni1); assertEquals("From: Suoritukset line: 60", 4, suoritukset.getLkm()); 
    suoritukset.lisaa(treeni1); assertEquals("From: Suoritukset line: 61", 5, suoritukset.getLkm()); 
    suoritukset.lisaa(treeni1); 
  } // Generated by ComTest END


  // Generated by ComTest BEGIN
  /** 
   * testKorvaaTaiLisaa78 
   * @throws SailoException when error
   * @throws CloneNotSupportedException when error
   */
  @Test
  public void testKorvaaTaiLisaa78() throws SailoException, CloneNotSupportedException {    // Suoritukset: 78
    Suoritukset suoritukset = new Suoritukset(); 
    Suoritus treeni1 = new Suoritus(), treeni2 = new Suoritus(); 
    treeni1.kirjaa(); treeni2.kirjaa(); 
    assertEquals("From: Suoritukset line: 85", 0, suoritukset.getLkm()); 
    suoritukset.korvaaTaiLisaa(treeni1); assertEquals("From: Suoritukset line: 86", 1, suoritukset.getLkm()); 
    suoritukset.korvaaTaiLisaa(treeni2); assertEquals("From: Suoritukset line: 87", 2, suoritukset.getLkm()); 
    Suoritus treeni3 = treeni1.clone(); 
    treeni3.taytaTreeniTiedoilla(); 
    Iterator<Suoritus> it = suoritukset.iterator(); 
    assertEquals("From: Suoritukset line: 91", true, it.next() == treeni1); 
    suoritukset.korvaaTaiLisaa(treeni3); assertEquals("From: Suoritukset line: 92", 2, suoritukset.getLkm()); 
    it = suoritukset.iterator(); 
    Suoritus treeni0 = it.next(); 
    assertEquals("From: Suoritukset line: 95", treeni3, treeni0); 
    assertEquals("From: Suoritukset line: 96", true, treeni0 == treeni3); 
    assertEquals("From: Suoritukset line: 97", false, treeni0 == treeni1); 
  } // Generated by ComTest END
}